prboom2/src/dsda/hud_components/weapon_text.c

```diff
@@ -82,6 +87,23 @@ void dsda_InitWeaponTextHC(int x_offset, int y_offset, int vpt, int* args, int a
   local = *data;
 
   local->grid = args[0];
+  if (arg_count > 1)
+    local->show_wpn = args[1];
+  else
+    local->show_wpn = true;
+
+  if (local->show_wpn) {
+    snprintf(local->label_w, sizeof(local->label_w), "%sW ", dsda_TextColor(dsda_tc_exhud_weapon_label));
```

Comment:
I'm not sure if it's currently possible, but it might be possible at some point to change the colors without reinitializing the hud, so we can't bake them in at this point.